sdk/src/logs/logger_provider.cc

```diff
@@ -0,0 +1,93 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "opentelemetry/sdk/logs/logger_provider.h"
+
+OPENTELEMETRY_BEGIN_NAMESPACE
+namespace sdk
+{
+namespace logs
+{
+LoggerProvider::LoggerProvider(std::shared_ptr<LogProcessor> processor) noexcept
+    : processor_{processor}
+{}
+
+LoggerProvider::LoggerProvider() noexcept : processor_{nullptr} {}
+
+opentelemetry::nostd::shared_ptr<opentelemetry::logs::Logger> LoggerProvider::GetLogger(
```

Comment:
If I read it correctly: we can allocate a named logger, but the logger object itself does not remember its name?

Reply:
Yes, this is following the spec https://github.com/open-telemetry/opentelemetry-specification/blob/master/specification/trace/api.md#get-a-tracer.

Reply:
Yup the logger's name is stored in the logger provider's [map of loggers](https://github.com/open-telemetry/opentelemetry-cpp/blob/774ec7f5a970df18cfcba450730716a7a2f5c5f2/sdk/include/opentelemetry/sdk/logs/logger_provider.h#L97)

Reply:
@reyang - this doesn't appear to be a logging spec?

Reply:
@reyang @MarkSeufert - can the logging provider still decorate events originating from logger with the logger name? Does it mean that you have to perform a search of logger in the map - what is the cost of that for each event emitted?

Reply:
To further elaborate on this in pseudocode:
```cpp
auto logger1 = lp->GetLogger("Storage");
auto logger2 = lp->GetLogger("Network");
auto logger3 = lp->GetLogger("Incoming");
auto logger4 = lp->GetLogger("Outgoing");
```

in most logging systems I can associate the logger with some system, submodule, class or component, means eventually I need to carry that information to display to developer. So developer can search / grep by logger name.

How is it done in our case if the logger itself does not know its name?

Reply:
> @reyang - this doesn't appear to be a logging spec?

Correct. The link above is for tracer.
The logger spec is still under construction, and based on the [data model](https://github.com/open-telemetry/opentelemetry-specification/blob/master/specification/logs/data-model.md#field-name), there is no such thing called `logger name` from consumption perspective.
If there is a valid scenario, we could consider to add it later but not now, and we will need to work on the spec first.

Reply:
@reyang - how is it done in Java, while mapping log4j API to OpenTelemetry? log4j is the most prominent Java Logging API (as discussed in several community meetings). log4j follows the model having named loggers:
https://logging.apache.org/log4j/1.2/manual.html

In my opinion, it is absolutely necessary to have ability to name a logger.

Reply:
Python cookbook - named loggers:
https://docs.python.org/3/howto/logging-cookbook.html

Reply:
https://github.com/open-telemetry/opentelemetry-specification/issues/1215

Reply:
Yea I agree this is important functionality for a logging library. It's not included in the spec right now, but once it is included we can go back and add it to the SDK.